nilearn/decomposition/canica.py

```diff
@@ -127,13 +128,18 @@ def __init__(self, mask=None, n_components=20, smoothing_fwhm=6,
             mask_strategy=mask_strategy, mask_args=mask_args,
             memory=memory, memory_level=memory_level,
             n_jobs=n_jobs, verbose=verbose)
+
+        if isinstance(threshold, float) and threshold > n_components:
+            raise ValueError("Threshold must not be higher than number "
+                             "of maps. "
+                             "Number of maps is %s and you provided %s" %
```

Comment:
Can you include threshold in the error message. I think writing as and "you provided threshold=%s" would be bit clear.